test/blockchaintest/blockchaintest_runner.cpp

```diff
@@ -222,15 +226,28 @@ void run_blockchain_tests(std::span<const BlockchainTest> tests, evmc::VM& vm)
                 bytes32{});
         EXPECT_EQ(c.genesis_block_header.logs_bloom, bytes_view{state::BloomFilter{}});
 
-        auto state = c.pre_state;
-
         TestBlockHashes block_hashes{
             {c.genesis_block_header.block_number, c.genesis_block_header.hash}};
-        const auto* parent_header = &c.genesis_block_header;
+
+        struct BlockData
+        {
+            const BlockHeader* header;
+            TestState state;
+            intx::uint256 total_difficulty;
+        };
+        std::unordered_map<hash256, BlockData> block_data{{{c.genesis_block_header.hash,
```

Comment:
`block_datas` or `block_datas_by_hash` maybe?

Reply:
data is already plural (from datum :see_no_evil:)

Reply:
yeah, maybe it's fine as is, nvm